chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/BrokenSurveillanceRulesCreatorJob.java

```diff
@@ -287,34 +288,34 @@ private BrokenSurveillanceRulesDTO addNcData(BrokenSurveillanceRulesDTO rule, Su
         } else {
             rule.setNonconformityCriteria(nc.getNonconformityType());
         }
+        LocalDateTime ncCloseDate = null;
+        if (nc.getNonConformityCloseDate() != null) {
+            ncCloseDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getNonConformityCloseDate().getTime()), ZoneId.systemDefault());
+            rule.setDateOfDeterminationOfNonconformity(dateFormatter.format(ncCloseDate));
+        }
         LocalDateTime ncDeterminationDate = null;
         if (nc.getDateOfDetermination() != null) {
-            ncDeterminationDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getDateOfDetermination().getTime()),
-                    ZoneId.systemDefault());
+            ncDeterminationDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getDateOfDetermination().getTime()), ZoneId.systemDefault());
             rule.setDateOfDeterminationOfNonconformity(dateFormatter.format(ncDeterminationDate));
         }
         LocalDateTime capApprovalDate = null;
         if (nc.getCapApprovalDate() != null) {
-            capApprovalDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapApprovalDate().getTime()),
-                    ZoneId.systemDefault());
+            capApprovalDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapApprovalDate().getTime()), ZoneId.systemDefault());
             rule.setCorrectiveActionPlanApprovedDate(dateFormatter.format(capApprovalDate));
         }
         LocalDateTime capStartDate = null;
         if (nc.getCapStartDate() != null) {
-            capStartDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapStartDate().getTime()),
-                    ZoneId.systemDefault());
+            capStartDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapStartDate().getTime()), ZoneId.systemDefault());
             rule.setDateCorrectiveActionBegan(dateFormatter.format(capStartDate));
         }
         LocalDateTime capMustCompleteDate = null;
         if (nc.getCapMustCompleteDate() != null) {
-            capMustCompleteDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapMustCompleteDate().getTime()),
-                    ZoneId.systemDefault());
+            capMustCompleteDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapMustCompleteDate().getTime()), ZoneId.systemDefault());
             rule.setDateCorrectiveActionMustBeCompleted(dateFormatter.format(capMustCompleteDate));
         }
         LocalDateTime capEndDate = null;
         if (nc.getCapEndDate() != null) {
-            capEndDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapEndDate().getTime()),
-                    ZoneId.systemDefault());
+            capEndDate = LocalDateTime.ofInstant(Instant.ofEpochMilli(nc.getCapEndDate().getTime()), ZoneId.systemDefault());
```

Comment:
The conversion of Long to LocalDateTime happens a number of times in here. I think in my OCD-3659 PR I added a function to DateUtil to do this because I was doing the same thing going from MUU date (Long) to PIU date (LocalDate) and vice versa. Not helpful at this time but just a note we may want to refactor places to use those utility methods later.

Reply:
👍